### PR TITLE
Refactor sparsity utils.py

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
Move sparsity utils to new way collecting wrapped layers and operations

### CODE REVIEW

File: nncf/tensorflow/sparsity/utils.py [hunk 1]
```diff
@@ -19,53 +19,42 @@
...
+                    weights_attr_name=weight_attr,
+                    operation_name=op.name)
+            ))
 
     return TFModelTransformer(model).transform(transformations)
 
 
-def apply_fn_to_op_weights(model: tf.keras.Model, op_names: List[str], fn = lambda x: x):
-    sparsifyed_layers = collect_wrapped_layers(model)
+def apply_fn_to_op_weights(model: tf.keras.Model, op_names: List[str], fn=lambda x: x):
```
Comment (andrey-churkin) [inline nncf/tensorflow/sparsity/utils.py:45]: Seems like this method is redundant. I suggest removing it and refactoring this code
https://github.com/openvinotoolkit/nncf/blob/0389d232cdefd7bc50f6d9adc285555f5947c254/nncf/tensorflow/sparsity/rb/algorithm.py#L102
as following
```python
target_ops = []
for wrapped_layer, _, op in get_nncf_operations(target_model, op_names):
    target_ops.append(
        (op, wrapped_layer.get_operation_weights(op.name))
    )
```

**Review by andrey-churkin — CHANGES_REQUESTED** [review]

File: nncf/tensorflow/sparsity/utils.py (hunk 1 above)
Comment (daniil-lyakhov) [reply nncf/tensorflow/sparsity/utils.py]: Done, please check

File: nncf/tensorflow/sparsity/utils.py [hunk 2]
```diff
@@ -19,52 +19,33 @@
...
-        for ops in layer.weights_attr_ops.values():
-            for op in ops.values():
-                if op.name in op_names:
-                    weight = layer.get_operation_weights(op.name)
-                    target_ops.append((op, fn(weight)))
-    return target_ops
-
-
-def apply_mask(wrapped_layer: NNCFWrapper, weight_attr: str, op_name: str):
+def apply_mask(wrapped_layer: NNCFWrapper, weight_attr: str, op: NNCFOperation):
```
Comment (alexsu52) [inline nncf/tensorflow/sparsity/utils.py:45]:
```suggestion
def apply_mask(wrapped_layer: NNCFWrapper, weight_attr: str, op: NNCFOperation) -> None:
```

**Review by alexsu52 — APPROVED** [review]

File: nncf/tensorflow/sparsity/utils.py (hunk 2 above)
Comment (daniil-lyakhov) [reply nncf/tensorflow/sparsity/utils.py]: Done

**Review by andrey-churkin — APPROVED** [review]

**Comment by alexsu52** [comment]
Jenkins please retry a build

**Pull request merged**